src/app/components_ngrx/delete-work-item/delete-work-item.component.ts

```diff
@@ -0,0 +1,55 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { Store } from '@ngrx/store';
+import { Observable } from 'rxjs';
+import { first } from 'rxjs/operators';
+import { Delete } from '../../actions/work-item.actions';
+import { PermissionQuery } from '../../models/permission.model';
+import { WorkItemUI } from '../../models/work-item';
+import { ModalService } from '../../services/modal.service';
+import { AppState } from '../../states/app.state';
+
+@Component({
+  selector: 'f8-delete-workitem',
+  templateUrl: './delete-work-item.component.html',
+  styleUrls: ['./delete-work-item.component.less']
+})
+
+export class DeleteWorkItemComponent {
+
+  @Input('workItem') set workItemInput(val: WorkItemUI) {
+    if (val) {
+      this.workItem = val;
+      this.allowDelete =
```

Comment:
But if we do it like how you are suggesting then we have to write this same code at more than one place and since this delete work item component is deeply nested for eg delete-work-item is a child of work-item-cell and work-item-cell is a child of query component and work-item-cell also might be a pure component so I have to add this in query component and then subscribe to it in the work-item-cell component and since in the work-item-cell we have the direct reference for the current workitem but not in the query component as it sends the list of workitems in the datatable so if I want to have the current work-item in the query component then I will have to send it back from the work-item-cell component and that doesnt make sense. So either we can keep this in the delete work item component or in the work-item-cell component.